src/main/java/duke/pack/AddCommand.java

```diff
@@ -0,0 +1,35 @@
+package duke.pack;
+
+/**
+ * Represents a command to add a task.
+ */
+public class AddCommand extends Command {
+    protected Task task;
+
+    public AddCommand(Task task) {
+        this.task = task;
+    }
+
+    /**
+     * Executes the add command and adds task to the TaskList.
+     * @param tasks TaskList containing all the added tasks
+     * @param ui UI that handles interactions with user
+     * @param storage Storage that handles updating of tasks in hard disk
+     * @throws DukeException if task cannot be saved to hard disk
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        tasks.addTask(task);
+        storage.save(tasks);
+        ui.showAdd(task);
+        ui.showCount(tasks);
+    }
+
+    /**
+     * indicates whether command is exit
```

Comment:
Hi Cheyanne, thank you very much for the feedback! :) I'm not sure how to reply to your overall comment above so I shall reply here. By unnecessary files do you mean those in .idea and bin etc? I didn't do anything in particular, so I'm not sure why they weren't pushed but I think it's possible to add files to .gitignore to prevent them from showing up after pushing? Hope that helps! :)